### PR TITLE
[dashboard] Add error type and error message to dashboard for task tables

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

This PR adds task error info to the dashboard (e.g. the error type and the exact error trace)

We will also see system errors details as well as applications errors

We will now be able to see OOM error detail for a task if it was killed:
image

- Cancelled task that are cancelled before running: 
image

- Actor tasks failed due to actor killed:
image

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
this is on top of the other PR right?

File: dashboard/client/src/components/TaskTable.tsx [hunk 1]
```diff
@@ -253,6 +255,9 @@ const TaskTable = ({
                     )}
                   </TableCell>
                   <TableCell align="center">{func_or_class_name}</TableCell>
+                  <TableCell align="center">
+                    {error_type ? error_type : "-"}
+                  </TableCell>
```
Comment (alanwguo) [inline dashboard/client/src/components/TaskTable.tsx:260]: Do we need an extra column? I already have a button to view error details.

What is "error type" anyways?

**Review by alanwguo — COMMENTED** [review]
Are we trying to merge this by today? I don't have full context of the change so probably it would be good to sync for 15 minutes to understand what this is trying to do.

**Comment by rickyyx** [comment]
> this is on top of the other PR right?

oh yes, so probably let's wait for that to be merged, and this should be much easier to review.

File: dashboard/client/src/components/TaskTable.tsx (hunk 1 above)
Comment (rickyyx) [reply dashboard/client/src/components/TaskTable.tsx:260]: It's shot code for task errors, e.g. WORKER_DIED, ACTOR_DIED, see some examples here (not all are related to tasks): https://github.com/ray-project/ray/blob/20a6226fe0c95e223fbab0e68ff5ef81f2f84c9c/src/ray/protobuf/common.proto#L141


But yes, we could put the error type in the Error details pop-up.

File: dashboard/client/src/type/task.ts [hunk 2]
```diff
@@ -39,6 +39,8 @@ export type Task = {
   end_time_ms: number | null;
   worker_id: string | null;
   profiling_data: ProfilingData;
+  error_type: string | null;
+  error_message: string | null;
```
Comment (rkooo567) [inline dashboard/client/src/type/task.ts:43]:
```suggestion
  error_message: string;
```

this is always there right?

File: python/ray/experimental/state/common.py [hunk 3]
```diff
@@ -1439,7 +1439,16 @@ def protobuf_to_task_state_dict(message: TaskEvents) -> dict:
     if latest_state == "FAILED":
         error_info = state_updates.get("error_info", None)
         if error_info:
-            task_state["error_message"] = error_info.get("error_message", "")
+            task_state["error_message"] = remove_ansi_escape_codes(
```
Comment (rkooo567) [inline python/ray/experimental/state/common.py:1442]: can you write a comment about example?

**Comment by rkooo567** [comment]
cc @alanwguo to review

**Comment by ollie-iterators** [comment]
There are some areas that could be fixed by merging recent changes into the branch

File: dashboard/client/src/type/task.ts (hunk 2 above)
Comment (rickyyx) [reply dashboard/client/src/type/task.ts:43]: I think they should be null when no errors happen: https://github.com/ray-project/ray/blob/d21b8326103a2c65c9130c7d82c8a2a4bb4869ed/python/ray/experimental/state/common.py#L588-L591

And `error_message` also depends on `error_type`: if no error_type, there shouldn't be error_message.

**Review by alanwguo — APPROVED** [review]
lgtm!

**Pull request merged**